File: prototype/base/src/main/java/com/project/base/repository/AppUserRepository.java
package com.project.base.repository;

import com.project.base.domain.AppUser;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Mono;

public interface AppUserRepository extends ReactiveCrudRepository<AppUser, Long> {

    Mono<AppUser> findByUsername(String username);

    Mono<Boolean> existsByUsername(String username);
}

File: prototype/lietzsche/src/main/kotlin/com/stxtory/lietzsche/repository/AppUserRepository.kt
package com.stxtory.lietzsche.repository

import com.stxtory.lietzsche.domain.AppUser
import org.springframework.data.repository.kotlin.CoroutineCrudRepository

interface AppUserRepository : CoroutineCrudRepository<AppUser, Long> {
    suspend fun findByUsername(username: String): AppUser?
    suspend fun existsByUsername(username: String): Boolean
}